### PR TITLE
Implement checks for vmap escaped errors

### PR DESCRIPTION
**Pull request by srossross**
Follow on to https://github.com/pytorch/pytorch/pull/89077

### CODE REVIEW

**Comment by srossross** [comment]
@kshitij12345 do you think it is worth while spending the time to implement new tests for each of these changes?

**Comment by kshitij12345** [comment]
I think that would make sense (maybe an OpInfo test to confirm all the ops throw an error). However, let's wait for @zou3519 to confirm (as I don't know much about this part of the code).

Thanks!

File: test/functorch/test_vmap.py [hunk 1]
```diff
@@ -3939,6 +3941,15 @@ def f(x):
...
 
+        with self.assertRaisesRegex(RuntimeError, common_message.format("binary_cross_entropy_plumbing")):
+            torch.nn.functional.binary_cross_entropy(escaped, torch.zeros([3, 3, 3, 3]))
+
+        with self.assertRaisesRegex(RuntimeError, common_message.format("boxed_existing_bdim_all_batch_rule")):
+            torch.nn.functional.adaptive_max_pool2d(escaped, output_size=(1, 1))
+
+        with self.assertRaisesRegex(RuntimeError, common_message.format("boxed_reduction_batch_rule")):
+            escaped.argmin()
+
```
Comment (srossross) [inline test/functorch/test_vmap.py:3952]: I couldn't find a way to invoke `boxed_all_tensors_have_optional_bdim` it looks like it is only used here:

```
  ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED_CONTIG1(3, adaptive_max_pool2d_backward, 2);
  ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED_CONTIG1(4, adaptive_max_pool3d_backward, 2);
  ALL_TENSORS_HAVE_OPTIONAL_BDIM_BOXED_CONTIG1(3, max_pool2d_with_indices_backward, 2);
```

is there a way to call the backward version directly on the `escaped` tensor?

Comment (zou3519) [reply test/functorch/test_vmap.py:3952]: It's possible to do torch.ops.aten.adaptive_max_pool2d_backward

**Review by zou3519 — APPROVED** [review]

File: test/functorch/test_vmap.py (hunk 1 above)
Comment (srossross) [reply test/functorch/test_vmap.py:3952]: added in 3a74733db81f74383605f514c4e41133be857b4b

**Pull request closed without merging**